docs/ref/models/querysets.txt

```diff
@@ -2431,10 +2431,12 @@ This has a number of caveats though:
 * The model's ``save()`` method will not be called, and the ``pre_save`` and
   ``post_save`` signals will not be sent.
 * It does not work with child models in a multi-table inheritance scenario.
-* If the model's primary key is an :class:`~django.db.models.AutoField`, the
-  primary key attribute can only be retrieved on certain databases (currently
-  PostgreSQL, MariaDB, and SQLite 3.35+). On other databases, it will not be
-  set.
+* To retrieve the primary key attribute, you need to meet all these conditions:
+    * The model's primary key is an :class:`~django.db.models.AutoField`, or 
+       its subclasses.
+    * The database has ``can_return_rows_from_bulk_insert=True`` (currently
+       PostgreSQL, MariaDB, and SQLite 3.35+). 
```

Comment:
Users don't need to be aware of database backend feature flags, so I'd prefer to add a `ignore_on_conflict` note to the first sentence, e.g.

_"* If the model's primary key is an :class:`~django.db.models.AutoField` and `ignore_on_conflict` is False, the primary key .."_

Reply:
On the other hand, it's already documented that:

_"Enabling the ignore_conflicts or update_conflicts parameter disable setting the primary key on each model instance (if the database normally support it)."_

Reply:
Apologies for the late reply. was busy wth client work.

I prefer that the condition about enabling ignore_conflicts parameter disable setting the primary key to be together with the condition about the model's primary key being a class of AutoField instead of separate lines. 

Even if there's repetition, I think it's better this way.

Also, based on what I read in the code, setting update_on_conflict doesn't affect the setting of primary key.

https://github.com/django/django/blob/134cde8415b3a9118262ef4ea34cf798c258be03/django/db/models/query.py#L1870C69-L1870C69

this line literally says `on_conflict is None or on_conflict == OnConflict.UPDATE` the inserted_rows will be extended